### ISSUE TITLE
[android][auto] class file for com.google.common.util.concurrent.ListenableFuture not found  

### ISSUE THREAD

**Issue by rtsisyk** [organicmaps/organicmaps#6106]
```
android/app/src/main/java/app/organicmaps/background/OsmUploadWork.java:18: error: cannot access ListenableFuture
public class OsmUploadWork extends Worker
       ^
  class file for com.google.common.util.concurrent.ListenableFuture not found  
```

**Comment by Jean-BaptisteC** [comment]
https://stackoverflow.com/questions/64290141/android-studio-class-file-for-com-google-common-util-concurrent-listenablefuture

**Comment by rtsisyk** [comment]
@Jean-BaptisteC, yeah, I noticed that. The problem is that Guava is not really needed. `androidx.car.app:app` adds entire `com.google.guava:guava` for `com.google.common.util.concurrent.ListenableFuture` class and few usages of `com.google.common.collect.ImmutableSet`.  I found that `ListenableFuture` is already extracted into a separate `androidx.concurrent:concurrent-listenablefuture` library. However, a replacement for `ImmutableSet` is still need to be provided. It would be even better if Google can just remove it from the code of `androidx.car.app:app`.

I filed a ticket: https://issuetracker.google.com/issues/301671348

**Comment by soshial** [comment]
Did you find a replacement for `ImmutableSet` and `ListenableFuture`? Does it mean that `implementation 'com.google.guava:guava:33.2.0-android'` can be deleted from Gradle? @rtsisyk
